src/fs/fat.rs

```diff
@@ -0,0 +1,102 @@
+#[cfg(esp_idf_soc_sdmmc_host_supported)]
+use core::ffi::c_void;
+
+use crate::private::cstr::*;
+
+use crate::sd::host::SdDevice;
+use crate::{sd::host::SdHost, sys::*};
+
+pub struct FatBuilder {
+    mount_config: esp_vfs_fat_mount_config_t,
+    base_path: CString,
+}
+
+impl Default for FatBuilder {
+    fn default() -> Self {
```

Comment:
Let's get away from the `builder` pattern, as it is not used anywhere else in the `esp-idf-*` crates.
My suggestion:
- create a `config` public module
- Create (inside that module) a `Config` struct
- Put `max_files`, `format_if_mount_failed`, `allocation_unit_size`, etc., **except** `base_path` (see below why) in the `Config` struct
- Implement sane defaults for all parameters of the `Config` struct and besides arg-less `new`, implement `Default` for it. Thus, the user can just pass `&Config::new()` or `&Default::default()` if she does not care
- Please look at how these `Config` structs are used throughout `esp-idf-hal`
- Create a `new` constructor on the `Fat` structure which takes:
  * `SdHost`
  * `&config::Config` (our new structure)
  * `base_path` (for mounting) - this we should keep outside of `Config` as it does not have a sane default